Added lives

joppevanbasten opened the pull request:
added lives, The lives script has to be added to the Canvas.

joppevanbasten committed:
Added lives

added lives script, Add lives script to canvas

gwnvisje commented on Assets/Scripts/Health/Lives.cs:
```diff
@@ -0,0 +1,55 @@
+		[SerializeField] private int lives = 3;
+		[SerializeField] private int size = 15;
+		private Sprite heart;
+		List<GameObject> hearts;
+		RectTransform canvas;
```
access modifier pls.

gwnvisje commented on Assets/Scripts/Health/Lives.cs:
```diff
@@ -0,0 +1,55 @@
+			}
+		}
+
+		public int GetLives()
+		{
+			return hearts.Count;
+		}
```
niet heel bijzonder, maar maak hier een Property van. Is wat netter.

gwnvisje commented on Assets/Scripts/Health/Lives.cs:
```diff
@@ -0,0 +1,55 @@
+	public class Lives : MonoBehaviour
+	{
+		[SerializeField] private int lives = 3;
+		[SerializeField] private int size = 15;
```
wat als ik een size wil van 13.5 !!!!

joppevanbasten committed:
Cleaned up code

added acces modifier to hearts and canvas, size is float, currentLives is a property

gwnvisje commented on Assets/Scripts/Health/Lives.cs:
```diff
@@ -0,0 +1,56 @@
+	public class Lives : MonoBehaviour
+	{
+
+		public int currentLives
+		{
+			get { return hearts.Count; }
+		}
```
Properties moeten altijd met hoofdletter.

Dit kan trouwens ook korter (hoeft niet hoor, als je dit meer leesbaar vindt)
`public int CurrentLives => hearts.Count`

joppevanbasten replied:
fixed

gwnvisje commented on Assets/Scripts/Health/Lives.cs:
```diff
@@ -0,0 +1,56 @@
+
+			for (int i = 0; i < lives; i++)
+			{
+				GameObject g = new GameObject($"heart {i}");
+				g.AddComponent<Image>().sprite = heart;
+				RectTransform rTransform = g.GetComponent<RectTransform>();
+				g.transform.parent = transform;
```
noem een object geen g, geef het een leesbare naam zodat het makkelijker is om te begrijpen wat voor object het is.

gwnvisje commented on Assets/Scripts/Health/Lives.cs:
```diff
@@ -0,0 +1,56 @@
+			if (hearts.Count > 0)
+			{
+				Destroy(hearts[hearts.Count - 1]);
+				hearts.RemoveAt(hearts.Count - 1);
```
voeg misschien een event toe die andere classes kunnen gebruiken als health nul of minder is?

joppevanbasten committed:
Lives AutoScale

Scales the lives to the screen size

joppevanbasten committed:
Seperate class for ui lives

gwnvisje commented on Assets/Scripts/Health/Lives.cs:
```diff
@@ -0,0 +1,40 @@
+		{
+			if (CurrentLives > 0)
+			{
+				CurrentLives--;
```
Gooi dit buiten de if statement. De functie impliceert dat levens verloren worden, maar dit gebeurt alleen maar als het boven de nul is. Het is ook niet erg als er minder dan 0 levens zijn dus daar is een check voor.

gwnvisje commented on Assets/Scripts/Health/LivesUI.cs:
```diff
@@ -0,0 +1,49 @@
+		private Sprite heart;
+		private List<GameObject> hearts;
+		private RectTransform canvas;
+		Lives lives = new Lives();
```
access modifier

gwnvisje commented on Assets/Scripts/Health/LivesUI.cs:
```diff
@@ -0,0 +1,49 @@
+				rTransform.position = new Vector2(rTransform.rect.width + (size / 4 + rTransform.rect.width) * i, Screen.height - size);
+				hearts.Add(life);
+			}
+			lives.LifeLostEvent += OnLiveLostEvent;
```
Je unsubscribed dit nooit. Unsubscribe het in een `OnDestroy()` functie.

gwenvis reviewed (commented):
paar dingen nog

joppevanbasten committed:
unsuvscribed OnLivesLostEvent
